katalog/etcd-backup/etcd-backup-s3/cronjob.yaml

```diff
@@ -0,0 +1,105 @@
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: etcd-backup-s3
+spec:
+  schedule: "* * 2 * *"
+  successfulJobsHistoryLimit: 3
+  failedJobsHistoryLimit: 5
+  concurrencyPolicy: Allow
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          volumes:
+            - name: etcd-certs
+              hostPath:
+                path: /etc/etcd/pki/etcd
+            - name: persistence
+              emptyDir: {}
+            - name: host-backup
+              hostPath:
+                path: /tmp/etcd-backup
+            - name: etcd-backup-s3-rclone-conf
+              secret:
+                secretName: etcd-backup-s3-rclone-conf
+          initContainers:
+            - name: backupper
+              image: etcd
+              envFrom:
+                - configMapRef:
+                    name: etcd-backup-certificates-location
+              securityContext:
+                runAsUser: 0
+                runAsGroup: 0
+              command: ["/bin/bash", "-c"]
+              args:
+                - "etcdctl snapshot save /backup/fury-etcd-snapshot-$(date +'%Y%m%d%H%M').etcdb"
+              volumeMounts:
+                - mountPath: /etcd
+                  name: etcd-certs
+                  readOnly: true
+                - mountPath: /backup
+                  name: persistence
+            - name: move-to-s3
+              image: rclone
+              command:
+                - "/bin/sh"
+                - "-c"
+              args:
+                - 'rclone copy /etcd-backup "$TARGET"'
+              env:
+                - name: TARGET
+                  valueFrom:
+                    configMapKeyRef:
+                      name: etcd-backup-config
+                      key: target
+              volumeMounts:
+                - mountPath: /config/rclone
+                  name: etcd-backup-s3-rclone-conf
+                  readOnly: true
+                - mountPath: /etcd-backup
+                  name: persistence
+          containers:
+            - name: delete-old
+              image: rclone
+              command:
+                - "/bin/sh"
+                - "-c"
+              args:
+                - 'rclone delete "$TARGET" --min-age="$RETENTION" --include=fury-etcd-snapshot*.etcdb'
+              env:
+                - name: TARGET
+                  valueFrom:
+                    configMapKeyRef:
+                      name: etcd-backup-config
+                      key: target
+                - name: RETENTION
+                  valueFrom:
+                    configMapKeyRef:
+                      name: etcd-backup-config
+                      key: retention
+              volumeMounts:
+                - mountPath: /config/rclone
+                  name: etcd-backup-s3-rclone-conf
+                  readOnly: true
+          restartPolicy: OnFailure
+          hostNetwork: true
```

Comment:
Is hostNetwork required? we should be able to reach etcd even from a non privileged pod right?

Reply:
At the moment we're only backupping etcd servers which reside on the control-planes. In the next iteration of this feature we will also support backupping etcd server which reside on an external cluster (the new external etcd feature being merged right now). 

However, in case etcd is installed on control-planes only, the etcd server listens on `localhost` only (https://github.com/sighupio/fury-kubernetes-on-premises/blob/35b4b46177bfe7ed17d424007a4416c87effd093/roles/etcd/defaults/main.yml#L12), thus requiring the `hostNetwork: true`.

Reply:
We can handle the external `etcd` in `fury-distribution` thus leaving the `vanilla` package here as it is with no further modifications.